File: code-coaches/hard/security/security.rb
def is_money_safe(layout)
    # Find the indices of the money, thief, and guards on the casino floor
    money_index = layout.index("$")
    thief_index = layout.index("T")
    guard_indices = layout.chars.each_index.select { |i| layout[i] == "G" }

    # Check if there is a guard between the money and the thief
    return guard_indices.any? { |i| i > [money_index, thief_index].min && i < [money_index, thief_index].max }
end


def main
    layout = gets.chomp

    if is_money_safe(layout)
        puts "quiet"
    else
        puts "ALARM"
    end
end


if __FILE__ == $0
    main
end


# ALTERNATIVE VERSION:
#
# diagram = gets.chomp
# diagramSplitted = diagram.upcase!.split("$")
#
# isThiefOnLeft = true
# thiefPos = nil
# guardPos = nil
#
# diagramSplitted.each_with_index do |part, idx1|
# 	if part.include? "T"
# 		part.each_char.with_index do |c, idx2|
# 			if c == 'G'
# 				if !isThiefOnLeft && guardPos
# 					next
# 				else
# 					guardPos = idx2
# 				end
# 			elsif c == 'T'
# 				thiefPos = idx2
# 			end
# 		end
# 	elsif idx1 == 0
# 		isThiefOnLeft = false
# 		next
# 	end
# end
#
# isLeftRobbed = (!guardPos || guardPos < thiefPos) && isThiefOnLeft
# isRightRobbed = (!guardPos || guardPos > thiefPos) && !isThiefOnLeft
#
# if isLeftRobbed || isRightRobbed
# 	puts "ALARM"
# else
# 	puts "quiet"
# end

File: code-coaches/hard/security/security.py
def is_money_safe(layout: str) -> bool:
    # Find the indices of the money, thief, and guards on the casino floor
    money_index = layout.find("$")
    thief_index = layout.find("T")
    guard_indices = [i for i, char in enumerate(layout) if char == "G"]

    # Check if there is a guard between the money and the thief
    return any(
        min(money_index, thief_index) < guard_index < max(money_index, thief_index)
        for guard_index in guard_indices
    )


def main():
    layout = input()

    if is_money_safe(layout):
        print("quiet")
    else:
        print("ALARM")


if __name__ == "__main__":
    main()


# ALTERNATIVE VERSION:
#
# diagram = input()
# diagram_splitted = diagram.split('$')
#
# is_thief_on_left = True
# thief_pos = None
# guard_pos = None
#
# for idx1, part in enumerate(diagram_splitted):
# 	if 'T' in part:
# 		for idx2, c in enumerate(part):
# 			if c.upper() == 'G':
# 				if not is_thief_on_left and guard_pos is not None:
# 					continue
# 				else:
# 					guard_pos = idx2
# 			elif c.upper() == 'T':
# 				thief_pos = idx2
# 	elif idx1 == 0:
# 		# if the field is not in the first array (left), then we know that it is in the right
# 		is_thief_on_left = False
# 		continue
#
# is_left_robbed = (guard_pos is None or guard_pos < thief_pos) and is_thief_on_left
# is_right_robbed = (guard_pos is None or guard_pos > thief_pos) and not is_thief_on_left
#
# if is_left_robbed or is_right_robbed:
# 	print("ALARM")
# else:
# 	print("quiet")
